pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -52,17 +53,18 @@ const PostStartHookName = "priority-and-fairness-config-producer"
 func (p RESTStorageProvider) NewRESTStorage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) (genericapiserver.APIGroupInfo, bool, error) {
 	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(flowcontrol.GroupName, legacyscheme.Scheme, legacyscheme.ParameterCodec, legacyscheme.Codecs)
 
-	if apiResourceConfigSource.VersionEnabled(flowcontrolv1alpha1.SchemeGroupVersion) {
-		flowControlStorage, err := p.v1alpha1Storage(apiResourceConfigSource, restOptionsGetter)
-		if err != nil {
-			return genericapiserver.APIGroupInfo{}, false, err
-		}
-		apiGroupInfo.VersionedResourcesStorageMap[flowcontrolv1alpha1.SchemeGroupVersion.Version] = flowControlStorage
+	v1beta1FlowControlStorage, err := p.v1beta1Storage(apiResourceConfigSource, restOptionsGetter)
+	if err != nil {
+		return genericapiserver.APIGroupInfo{}, false, err
+	}
+	apiGroupInfo.VersionedResourcesStorageMap[flowcontrolapisv1beta1.SchemeGroupVersion.Version] = v1beta1FlowControlStorage
+	if apiResourceConfigSource.VersionEnabled(flowcontrolapisv1alpha1.SchemeGroupVersion) {
+		apiGroupInfo.VersionedResourcesStorageMap[flowcontrolapisv1alpha1.SchemeGroupVersion.Version] = v1beta1FlowControlStorage
```

Comment:
Is it right to put the beta storage in the alpha entry in this map?

Reply:
Yeah, the storage is the same for all.

Reply:
I think so. I remember @lavalamp mentioning that they're supposed to share the same storage?

Reply:
either rename the method to be version-neutral, or make a copy per version. having a single storage constructor means that if a new type was added to surface in the alpha version, it would simultaneously/accidentally also get surfaced in the beta version